backend/src/main/java/com/github/NicolaiKopka/web_page_services/main_page/MainPageService.java

```diff
@@ -22,7 +24,49 @@ public List<Movie> getPopularMovies() {
         return movieDBApiConnect.getPopularMovies();
     }
 
-    public void getSoundtrackOnSpotify() throws OAuthProblemException, OAuthSystemException {
-        spotifyApiConnect.getSpotifyStatusForMovie();
+    public StreamingStatusDTO getSoundtrackOnSpotify(String movieName) throws OAuthProblemException, OAuthSystemException {
+        StreamingStatusDTO streamingStatusDTO = new StreamingStatusDTO();
+        streamingStatusDTO.setMovieName(movieName);
+
+        try {
+            List<SpotifyAlbum> albums = spotifyApiConnect.getSpotifyListOfMovieAlbums(movieName);
+
+            SpotifyAlbum album = checkForExactSoundtrack(albums, movieName);
```

Comment:
`album` can be `null` and the next line would cause a `NullPointerException`. When you change `checkForExactSoundtrack` as described in one of the next comments you could rewrite this methode like this:
```java
public StreamingStatusDTO getSoundtrackOnSpotify(String movieName) throws OAuthProblemException, OAuthSystemException {
    StreamingStatusDTO streamingStatusDTO = new StreamingStatusDTO();
    streamingStatusDTO.setMovieName(movieName);

    List<SpotifyAlbum> albums = spotifyApiConnect.getSpotifyListOfMovieAlbums(movieName);

    checkForExactSoundtrack(albums, movieName).ifPresentOrElse(
            album -> {
                String link = album.getExternalURLs().getAlbumUrl();
                streamingStatusDTO.getStreamingServiceStatus().put("spotify", true);
                streamingStatusDTO.getAlbumLinks().put("spotify", link);
            },
            () -> streamingStatusDTO.getStreamingServiceStatus().put("spotify", false)
    );
        
    return streamingStatusDTO;
}
```